chrome/test/chrome-api-shim.js

```diff
@@ -12,15 +10,17 @@ var chrome = {};
   }
 
   chrome.storage = {};
-  chrome.storage.sync = {
-    "set": function(dict, callback) {
+  function Storage() {
+    var storage_ = {};
+
+    this.set = function(dict, callback) {
```

Comment:
You can put these on `Storage.prototype`, put `storage_` on the instance (e.g., change line #14 to `this.storage_ = {}`). That seems cleaner.